src/history/file_backed.rs

```diff
@@ -89,8 +91,8 @@ impl HistoryView for FileBackedHistory {
 
     fn forward(&mut self) {
         match self.query.clone() {
-            HistoryNavigationQuery::Normal => {
-                if (self.cursor as isize) < self.entries.len() as isize - 1 {
+            HistoryNavigationQuery::Normal(_) => {
+                if self.cursor < self.entries.len() {
```

Comment:
let's say we have a vector of length 3, then the index value `3` is invalid. Right? We only have 0, 1, 2.

So shouldn't this be `self.entries.len() - 1` and not `self.entries.len()`?

Reply:
We use the state `self.cursor == self.entries.len()` to denote we are not currently browsing the history. Thus this allows to leave the history and also fixes the issues of skipping the most recent entry on the first go. This is safe as `VecDeque::get()` returns `None` for the resulting out of bound access, which is desired to denote leaving the list of history entries.